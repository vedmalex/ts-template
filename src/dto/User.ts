export class User {
  get fullName(): string {
    return `name: ${this.userName} email: ${this.email}`
  }
  constructor(public userName: string, public email: string) {
    console.log('done')
  }
}
