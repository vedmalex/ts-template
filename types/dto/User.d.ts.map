{"version":3,"file":"User.d.ts","sourceRoot":"","sources":["../../src/dto/User.ts"],"names":[],"mappings":"AAAA,qBAAa,IAAI;IACI,QAAQ,EAAE,MAAM;IAAS,KAAK,EAAE,MAAM;gBAAtC,QAAQ,EAAE,MAAM,EAAS,KAAK,EAAE,MAAM;CAC1D"}