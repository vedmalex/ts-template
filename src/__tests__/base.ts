import 'jest'
import { User } from '../dto/User'
describe('it must', () => {
  it('works', () => {
    expect('done').toBe('done')
  })
})

describe('user is created', () => {
  it('creates User', () => {
    const user = new User('Alex', 'vedmalex[at]gmail.com')
    expect(user).toMatchSnapshot('user')
  })
})
