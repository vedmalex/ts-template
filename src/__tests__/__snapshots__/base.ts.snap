// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user is created creates User: user 1`] = `
User {
  "email": "vedmalex[at]gmail.com",
  "userName": "Alex",
}
`;
